src/pipeline.ts

```diff
@@ -266,27 +268,36 @@ export class GitHubWorkflow extends PipelineBase {
     // create one file and make one step
     const relativeToAssembly = (p: string) => path.posix.join(cdkoutDir, path.relative(path.resolve(cdkoutDir), p));
     const fileContents: string[] = ['set -x'].concat(assets.map((asset) => {
-      return `npx cdk-assets --path "${relativeToAssembly(asset.assetManifestPath)}" --verbose publish "${asset.assetSelector}"`;
+      // we need the jobId to reference the outputs later
+      this.assetHashMap[asset.assetId] = jobId;
+      return [
+        `npx cdk-assets --path "${relativeToAssembly(asset.assetManifestPath)}" --verbose publish "${asset.assetSelector}"`,
+        `echo '::set-output name=hash::${asset.assetId}'`,
```

Comment:
This might be fine, it kinda depends on the multiplicity of things. Does this still work if there are multiple assets per asset step? Maybe that never happens?

Reply:
when there are multiple assets per step, it is because the same asset is getting uploaded to different regions, but with the same hash. But yes, it does not make sense to set the output each time; once should be sufficient.